lib/cyclonedx/cocoapods/component.rb

```diff
@@ -74,13 +85,39 @@ def validate_attributes(name, version, type, group)
         raise ArgumentError, "#{type} is not valid component type (#{VALID_COMPONENT_TYPES.join('|')})"
       end
 
+      def parse_name(name)
+        purls = name.split('/')
+        purl_name = CGI.escape(purls[0])
+        subpath = if purls.length > 1
+                    "##{name.split('/').drop(1).map do |component|
+                      CGI.escape(component)
+                    end.join('/')}"
+                  else
+                    ''
+                  end
+        [purl_name, subpath]
+      end
+
       def missing(str)
         str.nil? || str.to_s.strip.empty?
       end
 
       def exists_and_blank(str)
         !str.nil? && str.to_s.strip.empty?
       end
+
+      def create_purl(name)
```

Comment:
This `create_purl` function and the `parse_name` function above seem to be identical.  Is this function called anywhere?

Reply:
The duplicate, dead code has been removed.